source.go

```diff
@@ -46,13 +46,14 @@ func (s *Source) Configure(ctx context.Context, cfg map[string]string) error {
 }
 
 func (s *Source) Open(ctx context.Context, pos sdk.Position) error {
+	sdk.Logger(ctx).Info().Bytes("position", pos).Msg("Opening a source...")
 	client, err := NewConsumer()
 	if err != nil {
 		return fmt.Errorf("failed to create Kafka client: %w", err)
 	}
 	s.Consumer = client
 
-	err = s.startFrom(pos)
+	err = s.startFrom(ctx, pos)
 	if err != nil {
 		return fmt.Errorf("couldn't start from position: %w", err)
 	}
```

Comment:
Now that I think about it, we don't have a need to store `lastPositionRead` anymore, we can remove it completely and just call `s.Consumer.StartFrom` directly from here.
```suggestion
	err = s.Consumer.StartFrom(s.Config, string(pos))
	if err != nil {
		return fmt.Errorf("couldn't start from position %v: %w", string(pos), err)
	}
```